### DISCUSSION TITLE
Get `first` working for `table` or `record`

### DISCUSSION THREAD

**Discussion by Abdillah** [nushell/nushell#10529] [Ideas]
Some Nushell commands seems to just work in a subset of types, which makes remembering them harder.
Some of them is first / skip / last, which to be expected in table format, but actually can't handle them.

Or am I missing something?

**Comment by fdncred** [comment]
You'll have to give an example. select, get, first, last, skip, take should all work fine in tables.

**Reply by Abdillah** [reply]
Ah, I'm sorry, with your assertion, `$env` is not a table isn't it?
I can make it work by `transpose` it first.

```
$env | first 5 
# ^ not working

$env | transpose | first 5
# ^ works
```

Then, is table need to be mentioned [in the documentation](https://www.nushell.sh/commands/docs/first.html#input-output-types)?

![image](https://github.com/nushell/nushell/assets/3030950/b4f59104-05ae-48fc-912e-e4535af9a704)

**Reply by fdncred** [reply]
`$env` returns a record, not a list, nor a table. I don't believe `first` works with records.
